routes/user.js

```diff
@@ -0,0 +1,296 @@
+      message: "address saved",
+      data: null,
+    });
+  });
+});
+
+app.post("/forgot-password", async function (req, res) {
+  var username = req.body.username;
+  user.findOne({ username: username }, async function (err, userDetails) {
+    if (userDetails && userDetails._id) {
+      const token = jwt.sign({ username }, process.env.TOKEN_SECRET);
+      var token_post = new AccessToken({
+        user_id: userDetails._id,
+        access_token: token,
+      });
+      token_post.save(function (err) {
+        if (err) {
+          res.status(500).json({
+            error: 0,
+            message: err.message,
+            data: null,
+          });
+        }
+        const msg = {
+          to: `${userDetails.email}`,
+          from: "ankit@excellencetechnologies.info",
+          subject: "Sending with SendGrid is Fun",
+          text: "and easy to do anywhere, even with Node.js",
+          html: `<a href="http://localhost:4000/user/verify-reset-password/${token}">Click to Reset Password</a>`,
+        };
+        Mail.send(msg)
+          .then(() => {
+            console.log("Email sent");
+          })
+          .catch((error) => {
+            console.error(error);
+          });
+        res.status(200).json({
+          error: 0,
+          message: "token saved",
+          data: tokenId,
+        });
+      });
+    } else {
+      res.status(500).json({
+        error: 1,
+        message: "user not found",
+        data: null,
+      });
+    }
+  });
+});
+
+app.post("/verify-reset-password/:token", async function (req, res, next) {
+  var forgetToken = req.params.token;
+  if (forgetToken == null)
+    return res.status(401).json({
+      error: 0,
+      message: "invakid token",
+      data: null,
+    });
+  AccessToken.findOne(
+    { access_token: forgetToken },
+    async function (err, tokenDetail) {
+      if (tokenDetail && tokenDetail._id) {
+        jwt.verify(
+          forgetToken,
+          process.env.TOKEN_SECRET,
+          (err, verifiedJwt) => {
+            if (err) {
+              res.status(500).json({
+                error: 0,
+                message: err.message,
+                data: null,
+              });
+            } else {
+              var newPassword = req.body.password;
+              const salt = bcrypt.genSalt();
+              const userPassword = bcrypt.hash(newPassword, salt);
+              user.updateOne(
+                { _id: userDetails._id },
+                { $set: { password: userPassword } },
+                { new: true }
+              );
+              res.status(200).json({
+                error: 0,
+                message: "password reset successfully",
+                data: null,
+              });
+              next();
+            }
+          }
+        );
+      } else {
+        res.status(500).json({
+          error: 1,
+          message: "token not matched or expired",
+          data: null,
+        });
+      }
+    }
+  );
+});
```

Comment:
how can you verify that password reset token should not use more then one time ?
and password reset token expiry should be 10 min only ?